rclcpp/include/rclcpp/node.hpp

```diff
@@ -929,7 +929,7 @@ class Node : public std::enable_shared_from_this<Node>
   RCLCPP_PUBLIC
   RCUTILS_WARN_UNUSED
   OnSetParametersCallbackHandle::SharedPtr
-  add_on_set_parameters_callback(OnParametersSetCallbackType callback);
+  add_on_set_parameters_callback(OnParametersSetCallbackType callback, bool allowRecursion = false);
```

Comment:
I don't like the idea of having an `allowRecursion`, we should either always support recursion or not at all.
nit

```suggestion
  add_on_set_parameters_callback(OnParametersSetCallbackType callback, bool allow_recursion = false);
```